src/vs/platform/telemetry/common/telemetryUtils.ts

```diff
@@ -42,6 +43,20 @@ export function combinedAppender(...appenders: ITelemetryAppender[]): ITelemetry
 
 export const NullAppender: ITelemetryAppender = { log: () => null, dispose: () => TPromise.as(null) };
 
+
+export class LogAppender implements ITelemetryAppender {
+
+	constructor(@ILogService private readonly _logService: ILogService) { }
+
+	dispose(): TPromise<any> {
+		return TPromise.as(undefined);
+	}
+
+	log(eventName: string, data: any): void {
+		this._logService.trace(`telemetry/${eventName}`, data);
```

Comment:
@jrieken  My concern on using `trace` for the telemetry events is that the logs will now get very noisy. This is due to the `data` part of the event having a lot of "common" properties.
Anybody using the log level of `trace` is sure to get annoyed by all the noise.

An alternative is to 
- Do what you did here, but log only the event name + data without the "common" parts of the data. This can help in the development/debugging process
- Have a dedicated output channel for the telemetry events with event name + all data. This can help in our attempt in being transparent.

Thoughts? cc @kieferrm 

Below is a comparison between all data and data without the common parts

![image](https://user-images.githubusercontent.com/16890566/42484323-394ee226-83a7-11e8-8f84-e5db7ea73a1d.png)


![image](https://user-images.githubusercontent.com/16890566/42484241-e033005a-83a6-11e8-8b5f-5c00b3959ead.png)

Reply:
Yeah, I am all in for a dedicated output channel or, even better, a dedicated log file which then shows as output channel. If we have dedicated channel we might also consider other places that send telemetry, like the main process and other windows.

Reply:
generally, don't feel obligated to stick to my changes/approch. it was just an experiment